src/app/data.service.spec.ts

```diff
@@ -0,0 +1,26 @@
+import { TestBed } from '@angular/core/testing';
+import { Observable, from } from 'rxjs';
+
+import { DataService } from './data.service';
+
+describe('DataService', () => {
+  let service: DataService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({});
+    service = TestBed.inject(DataService);
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('should send reformatted input to output when submit() is called', async () => {
```

Comment:
What should be tested is that that submit route was called and makes the post request you expect.  

from flask import current_app

with current_app.test_client() as client:
    response = client.patch('..../submit', input=yourinputstring )
    self.assertEqual(response.status_code, 200)